src/Components/SavedSearchAlert/Mutations/createAdvisoryOpportunity.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  createAdvisoryOpportunityMutation$data,
+  createAdvisoryOpportunityMutation,
+} from "__generated__/createAdvisoryOpportunityMutation.graphql"
+import { commitMutation, Environment, graphql } from "relay-runtime"
+
+interface MutationInput {
+  searchCriteriaID: string
+  message: string
+}
+
+export const createAdvisoryOpportunity = (
+  environment: Environment,
+  attributes: MutationInput
+): Promise<createAdvisoryOpportunityMutation$data> => {
+  return new Promise((resolve, reject) => {
+    commitMutation<createAdvisoryOpportunityMutation>(environment, {
+      mutation: graphql`
+        mutation createAdvisoryOpportunityMutation(
+          $input: createAdvisoryOpportunityMutationInput!
+        ) {
+          createAdvisoryOpportunity(input: $input) {
+            advisoryOpportunityOrError {
+              ... on createAdvisoryOpportunitySuccess {
+                advisoryOpportunity {
+                  internalID
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: {
+        input: {
+          ...attributes,
+        },
+      },
+      onCompleted: response => {
+        resolve(response)
```

Comment:
Will add the better error handling, where the response is introspected to see if there's any errors, and if so `reject`

Reply:
or rather, will update to use the preferred `useMutation` hook